core/trino-main/src/main/java/io/trino/execution/TableGetSplitDistribution.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.stats.Distribution.DistributionSnapshot;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import javax.annotation.concurrent.Immutable;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class TableGetSplitDistribution
```

Comment:
Any particular reason for having this class rather than `Map<PlanNodeId, DistributionSnapshot>` ?

Reply:
I've tried to be aligned with JSON output we have currently, where we don't use objects properties where the name is dynamic. Most of the time it is converted to list (for example `operatorToStats` or `tasks`).